basic_autonomy/src/helper_functions.cpp

```diff
@@ -65,18 +66,21 @@ namespace waypoint_generation
     int get_nearest_index_by_downtrack(const std::vector<lanelet::BasicPoint2d>& points, const carma_wm::WorldModelConstPtr& wm, double target_downtrack)
     {
         size_t best_index = points.size() - 1;
```

Comment:
`best_index` should be an `int` type to avoid rollover. If `points` is empty, the function should return `-1` to indicate that there is no nearest index.

Reply:
Good idea. Updated `best_index` to be an `int` and had an additional slight update to handle an empty `points` vector.